src/components/AuditLogsTable/AuditLogsTablePagination/AuditLogsTablePagination.tsx

```diff
@@ -0,0 +1,99 @@
+import type { PropsWithSpread } from "@canonical/react-components";
+import { Button, Icon, Pagination, Select } from "@canonical/react-components";
+import classNames from "classnames";
+import type { HTMLProps, OptionHTMLAttributes } from "react";
+import { useSelector } from "react-redux";
+
+import { useQueryParams } from "hooks/useQueryParams";
+import { getAuditEvents } from "store/juju/selectors";
+
+import { DEFAULT_LIMIT_VALUE, DEFAULT_PAGE } from "../consts";
+
+import "./_audit-logs-table-pagination.scss";
+
+export enum Label {
+  FIRST_PAGE = "Back to first page",
+}
+
+type Props = PropsWithSpread<
+  {
+    showLimit?: boolean;
+  },
+  HTMLProps<HTMLDivElement>
+>;
+
+const LIMIT_OPTIONS: OptionHTMLAttributes<HTMLOptionElement>[] = [
+  { label: "50/page", value: DEFAULT_LIMIT_VALUE },
+  { label: "100/page", value: 100 },
+  { label: "200/page", value: 200 },
+];
+
+const scrollToTop = () => {
+  document?.querySelector<HTMLDivElement>(".l-application")?.scrollTo(0, 0);
+};
+
+const AuditLogsTablePagination = ({
+  className,
+  showLimit,
+  ...props
+}: Props) => {
+  const auditLogs = useSelector(getAuditEvents);
+  const [queryParams, setQueryParams] = useQueryParams<{
+    limit: string | null;
+    panel: string | null;
+    page: string | null;
+  }>({
+    limit: null,
+    panel: null,
+    page: DEFAULT_PAGE,
+  });
+  const limit = Number(queryParams.limit);
+  const page = Number(queryParams.page);
+  return (
+    <div
+      className={classNames("audit-logs-table-pagination", className)}
+      {...props}
+    >
+      {showLimit ? (
+        <Select
+          defaultValue={DEFAULT_LIMIT_VALUE}
+          options={LIMIT_OPTIONS}
+          onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
+            setQueryParams(
+              { limit: e.target.value, page: null },
+              { replace: true }
+            );
+          }}
+        />
+      ) : null}
+      <Button
+        className="u-no-margin--right"
+        disabled={page === Number(DEFAULT_PAGE)}
+        hasIcon
+        onClick={() => {
+          setQueryParams({ page: null });
+          scrollToTop();
+        }}
+      >
+        <Icon className="p-icon--rotate-270" name="back-to-top">
+          {Label.FIRST_PAGE}
+        </Icon>
+      </Button>
```

Comment:
I added this button to go back to the first page as it seemed like this was an action I was wanting to take a number of times, and the only way to do it was refresh or click the Logs link in the side nav.

Reply:
There is a small issue with this button and the `Filter` button when the horizontal width of the page is small. Both of them will grow in height.